make-file-list.sh

```diff
@@ -144,62 +161,62 @@ test -z "$required" || {
 	grep -v 'database file for .* does not exist' <"$pacman_stderr" >&2
 }
 
-packages="mingw-w64-$ARCH-git mingw-w64-$ARCH-git-credential-manager
+packages="mingw-w64-$PACMAN_ARCH-git mingw-w64-$PACMAN_ARCH-git-credential-manager
 git-extra openssh $UTIL_PACKAGES"
 if test -z "$MINIMAL_GIT"
 then
-	packages="$packages mingw-w64-$ARCH-git-doc-html ncurses mintty vim nano
+	packages="$packages mingw-w64-$PACMAN_ARCH-git-doc-html ncurses mintty vim nano
 		winpty less gnupg tar diffutils patch dos2unix which subversion perl-JSON
-		mingw-w64-$ARCH-tk mingw-w64-$ARCH-connect git-flow docx2txt
-		mingw-w64-$ARCH-antiword mingw-w64-$ARCH-odt2txt ssh-pageant
-		mingw-w64-$ARCH-git-lfs mingw-w64-$ARCH-xz tig $GIT_UPDATE_EXTRA_PACKAGES"
+		mingw-w64-$PACMAN_ARCH-tk mingw-w64-$PACMAN_ARCH-connect git-flow docx2txt
+		mingw-w64-$PACMAN_ARCH-antiword mingw-w64-$PACMAN_ARCH-odt2txt ssh-pageant
+		mingw-w64-$PACMAN_ARCH-git-lfs mingw-w64-$PACMAN_ARCH-xz tig $GIT_UPDATE_EXTRA_PACKAGES"
 fi
 pacman_list $packages "$@" |
 
 grep -v -e '\.[acho]$' -e '\.l[ao]$' -e '/aclocal/' \
 	-e '/man/' -e '/pkgconfig/' -e '/emacs/' \
-	-e '^/usr/lib/python' -e '^/usr/lib/ruby' -e '^/mingw../lib/python' \
+	-e '^/usr/lib/python' -e '^/usr/lib/ruby' -e '^/(mingw|clang(arm)?)../lib/python' \
 	-e '^/usr/share/subversion' \
-	-e '^/etc/skel/' -e '^/mingw../etc/skel/' \
+	-e '^/etc/skel/' -e '^/(mingw|clang(arm)?)../etc/skel/' \
 	-e '^/usr/bin/svn' \
 	-e '^/usr/bin/xml.*exe$' \
 	-e '^/usr/bin/xslt.*$' \
 	-e '^/usr/bin/b*zmore$' \
-	-e '^/mingw../bin/.*zstd\.exe$' \
-	-e '^/mingw../share/doc/openssl/' \
-	-e '^/mingw../share/doc/gettext/' \
-	-e '^/mingw../share/doc/lib' \
-	-e '^/mingw../share/doc/pcre2\?/' \
-	-e '^/mingw../share/doc/git-doc/.*\.txt$' \
-	-e '^/mingw../share/doc/zstd/' \
-	-e '^/mingw../lib/gettext/' -e '^/mingw../share/gettext/' \
-	-e '^/usr/include/' -e '^/mingw../include/' \
+	-e '^/(mingw|clang(arm)?)../bin/.*zstd\.exe$' \
+	-e '^/(mingw|clang(arm)?)../share/doc/openssl/' \
+	-e '^/(mingw|clang(arm)?)../share/doc/gettext/' \
```

Comment:
@dscho after some testing, I found that this capturing group (or the non-capturing group alternative: `'^/(?:mingw|clang(arm)?)../share/doc/gettext/'`) only works when `-P` is specified in `grep`:

```
Regexp selection and interpretation:
  -E, --extended-regexp     PATTERN is an extended regular expression (ERE)
  -F, --fixed-strings       PATTERN is a set of newline-separated strings
  -G, --basic-regexp        PATTERN is a basic regular expression (BRE)
  -P, --perl-regexp         PATTERN is a Perl regular expression
  -e, --regexp=PATTERN      use PATTERN for matching
  -f, --file=FILE           obtain PATTERN from FILE
  -i, --ignore-case         ignore case distinctions
  -w, --word-regexp         force PATTERN to match only whole words
  -x, --line-regexp         force PATTERN to match only whole lines
  -z, --null-data           a data line ends in 0 byte, not newline
```

However, when I tested with `-P`, I ran into the following error:

```
grep: the -P option only supports a single pattern
```

🤦🏼‍♂️ this is slightly getting out of hand.

Would it also be ok for you to simply add the `MSYSTEM_LOWER` variable to the pattern, like this:

```
-e "^/$MSYSTEM_LOWER/share/doc/gettext/" \
```

... or do you really need `mingw..` so that it matches both `mingw32` and `mingw64`? Just checking!

Reply:
I _guess_ that `$MSYSTEM_LOWER` would work. _Except_ in the case of `clangarm64` where we expect to also have to handle `mingw64`... But why not simply `-e "^/\(mingw\|clang\)[^/]*/share/doc/gettext/"`, i.e. match any top-level directory starting either with `mingw` or with `clang`?